internal/helm/repository/chart_repository.go

```diff
@@ -281,27 +284,57 @@ func (r *ChartRepository) DownloadChart(chart *repo.ChartVersion) (*bytes.Buffer
 }
 
 // CacheIndex attempts to write the index from the remote into a new temporary file
-// using DownloadIndex, and sets Path and cached.
-// The caller is expected to handle the garbage collection of Path, and to
+// using DownloadIndex. The original file from the server is written to OriginalPath
+// and the index is converted to JSON and written to Path. The cached field is also set.
+// The caller is expected to handle the garbage collection of Path and OriginalPath, and to
```

Comment:
As we [always load the index once we detect a divergence](https://github.com/fluxcd/source-controller/blob/4b64dd553f81dfe646949c2c89598e37cd6fe53e/internal/controller/helmrepository_controller.go#L492), maintaining the two files shouldn't be required.

Instead, once loaded, we can simply write it as JSON using `json.Encode` (which is now a cheap operation due to being JSON).